pep-0702.rst

```diff
@@ -79,15 +79,17 @@ Several users have requested support for such a feature:
 Specification
 =============
 
-The new ``@typing.deprecated()`` decorator can be used on a class, function or method.
+A new decorator ``@deprecated()`` is added to the :mod:`typing` module. This
+decorator can be used on a class, function or method to mark it as deprecated.
 With overloaded functions, the decorator may be applied to individual overloads,
 indicating that the particular overload is deprecated. The decorator may also be
 applied to the overload implementation function, indicating that the entire function
 is deprecated.
 
 The decorator takes a single argument of type ``str``, which is a message that should
 be shown by the type checker when it encounters a usage of the decorated object.
-The message must be a string literal or a string with a ``Literal`` type.
+The message must be a string literal or be statically resolvable to a single string
+(e.g., a :data:`typing.Final` variable of type ``str``).
```

Comment:
Can't say I believe this is the right constraint either. It's totally possible to have something of type `Final[str]` whose value cannot be determined by the type checker -- that type just means that it cannot be reassigned, but the initial value could still be dynamically computed.

Reply:
What about

```suggestion
(e.g., a :data:`~typing.Final` variable of type ``str`` assigned a string literal).
```

Reply:
Presumably for pyright that wouldn't be a problem, since it already tracks literal values. In mypy this will probably require something new, since it doesn't do this. No idea about Pyre, pytype and others. Maybe we should leave this discussion open for the more public debate of the PEP's merits and details. A base position to retreat to might be that it has to be a string literal? PEP 675 (`LiteralString`) comes to mind.

Reply:
I'll just go back to only string literals; type checkers can relax the requirements if they wish.